core/src/main/scala/design/hamu/job/EncodeJob.scala

```diff
@@ -0,0 +1,38 @@
+package design.hamu.job
+
+import design.hamu.model.{EncodeInput, EncodeOutput}
+
+import io.circe.Json
+import io.circe.parser
+import io.circe.syntax._
+import scala.collection.{mutable => m}
+
+object EncodeJob extends Job {
+
+  def apply(string: String): Either[Throwable, Json] =
+    parser
+      .parse(string) // returns Either[Throwable, Json] based on if string is valid json
+      .flatMap { json: Json =>
+        json.as[EncodeInput] // returns Either[Throwable, EncodeInput] based on if json can be decoded to EncodeInput
+      }
+      .map { input: EncodeInput =>
+        run(input).asJson // creates output and encode into json
+      }
+
+  def run(input: EncodeInput): EncodeOutput = {
+    val (output, uniqueChar) = convertToOutput(input.input)
+    EncodeOutput(output, uniqueChar)
+  }
+
+  def convertToOutput(inputArray: String): (String, Int) = {
+    (countContiguousChars(inputArray.head, inputArray, 0), inputArray.distinct.length)
+  }
+
+  def countContiguousChars(char: Char, inputArray: String, count: Int): String = {
+    if (inputArray.isEmpty) s"$char${count.toString}"
+    else {
+      if (char.equals(inputArray.head)) countContiguousChars(char, inputArray.reverse.dropRight(1).reverse, count + 1)
```

Comment:
So below is gonna be 2 traversal's through the string. Twice to reverse O(n)
`inputArray.reverse.dropRight(1).reverse`

You really want `tail` here to just get the tail of a `string` which will get you the same thing for O(1)